src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> node = first; node != null; node = node.next) {
+            if ((object != null && object.equals(node.item))
+                    || (object == null && node.item == null)) {
+                unlink(node);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkElementIndex(int index) {
+        if (!isElementIndex(index)) {
+            throw new IndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private String outOfBoundsMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+    private boolean isElementIndex(int index) {
+        return index >= 0 && index < size;
+    }
+
+    private Node<T> getNode(int index) {
+        checkElementIndex(index);
+        Node<T> node;
+        if (index < (size / 2)) {
+            node = first;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+        } else {
+            node = last;
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+        }
+        return node;
+    }
+
+    T unlink(Node<T> nodeToUnlink) {
+        final T element = nodeToUnlink.item;
+        final Node<T> next = nodeToUnlink.next;
+        final Node<T> prev = nodeToUnlink.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            nodeToUnlink.prev = null;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            nodeToUnlink.next = null;
+        }
+
+        nodeToUnlink.item = null;
+        size--;
+        return element;
+    }
+
+    void linkLast(T value) {
+        final Node<T> oldLast = last;
+        final Node<T> newNode = new Node<>(oldLast, value, null);
+        last = newNode;
+        if (oldLast == null) {
+            first = newNode;
+        } else {
+            oldLast.next = newNode;
+        }
+        size++;
+    }
+
+    void linkBefore(T value, Node<T> beforeThis) {
+        final Node<T> oldPrev = beforeThis.prev;
+        final Node<T> newNode = new Node<>(oldPrev, value, beforeThis);
+        beforeThis.prev = newNode;
+        if (oldPrev == null) {
+            first = newNode;
+        } else {
+            oldPrev.next = newNode;
+        }
+        size++;
+    }
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        private Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
     }
```

Comment:
fix this
<img width="728" alt="image" src="https://github.com/mate-academy/jv-linked-list/assets/81781479/db3dba20-3271-464e-8028-38a0da451b31">